src/v/pandaproxy/rest/rest_request_auth.cc

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright 2022 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#include "pandaproxy/rest/rest_request_auth.h"
+
+#include "config/configuration.h"
+#include "pandaproxy/logger.h"
+#include "seastar/http/exception.hh"
+#include "security/scram_algorithm.h"
+
+namespace pandaproxy {
+
+rest_request_authenticator::rest_request_authenticator(
+  cluster::controller* controller)
+  : _controller(controller) {}
+
+credential_t
+rest_request_authenticator::authenticate(const ss::httpd::request& req) {
+    if (_controller == nullptr) {
+        throw ss::httpd::server_error_exception("No controller given");
+    }
+
+    credential_t user = default_user();
+    // Throws httpd exception if header parse fails
+    user = parse_auth_header(req);
+
+    // Throws httpd exception if validation fails
+    // Pass otherwise
+    const auto& cred_store = _controller->get_credential_store().local();
+    validate_credentials(user, cred_store);
+
+    return user;
+}
+
+credential_t
+rest_request_authenticator::parse_auth_header(ss::httpd::request const& req) {
+    ss::sstring auth_hdr{req.get_header("authorization")};
+
+    if (auth_hdr.substr(0, 5) == "Basic") {
```

Comment:
This should be refactored to share code with the existing `request_auth` class (relocate it to a shared location if necessary) rather than duplicating.

Reply:
I'll use the request_authenticator then since it does everything I need except fetch the password.